app/(main)/settings/account/actions.ts

```diff
@@ -103,3 +106,47 @@ export async function disconnectGoogleIdentity() {
 export async function disconnectGitHubIdentity() {
 	return disconnectIdentity("github");
 }
+
+export async function getAccountInfo() {
+	const supabaseUser = await getUser();
+
+	const _users = await db
+		.select({ displayName: users.displayName, email: users.email })
+		.from(users)
+		.innerJoin(
+			supabaseUserMappings,
+			eq(users.dbId, supabaseUserMappings.userDbId),
+		)
+		.where(eq(supabaseUserMappings.supabaseUserId, supabaseUser.id));
+
+	return _users[0];
+}
+
+export async function updateDisplayName(formData: FormData) {
+	try {
+		const supabaseUser = await getUser();
+
+		if (!supabaseUser) {
+			throw new Error("User not found");
+		}
+
+		const displayName = formData.get("displayName") as string;
+
+		const userDbIdSubquery = db
+			.select({ userDbId: supabaseUserMappings.userDbId })
+			.from(supabaseUserMappings)
+			.where(eq(supabaseUserMappings.supabaseUserId, supabaseUser.id));
+
+		await db
+			.update(users)
+			.set({ displayName })
+			.where(eq(users.dbId, userDbIdSubquery));
```

Comment:
The reason I used this subquery is that drizzle's `eq` does not support fields containing `null`.
By the way, if `users.email` is `not null`, I could just write `eq(users.email, supabaseUser.email)`.

Reply:
This upcoming feature in drizzle might be useful for our case:
https://github.com/drizzle-team/drizzle-orm/pull/2963

However, our current implementation using subqueries is sufficient for our needs 👍